### DISCUSSION TITLE
Notes to instructors

### DISCUSSION THREAD

**Discussion by siwelwerd** [TeamBasedInquiryLearning/precalculus#147] [General]
Since we are making an instructor version that contains `<answer>`s (see #110), we discussed today if we could possibly have facilitation notes included in the instructor version as well.  Once #132 is implemented, any tag with the `@component="instructor"` attribute will be included only in the instructor version, but the question was raised as to how an instructor reading the instructor version could visually distinguish which things were only in the instructor version, versus in the student-facing version as well.

@StevenClontz is there a particular tag we could use for these?  One option is to put them all in `<aside>`s.  I was also wondering about `<note>` or `<commentary>`, but not sure how visually distinctive those will be -- maybe we could repurpose one to display 'Instructor Note' or similar, but I don't know how to accomplish that.

**Comment by StevenClontz** [comment]
My hot take is `<paragraphs component="instructor"><title>Note to Instructor</title>[content]</paragraphs>`. In particular, the `<paragraphs/>` element is unnumbered, so numbering won't be impacted if these are shown/hidden.

This does seem a bit boilerplatey though...

**Reply by siwelwerd** [reply]
*Of course* PreTeXt has `<p>` for paragraph and `<paragraphs>` for... paragraphs. 🤦🏻‍♂️

**Comment by StevenClontz** [comment]
![image](https://github.com/TeamBasedInquiryLearning/precalculus/assets/1559632/fe712b7d-f518-4b27-bdd6-7c6043d5b897)

**Reply by StevenClontz** [reply]
I think I like `<commentary/>` - I would still add `<title>Instructor Note</title>` to each.

**Reply by StevenClontz** [reply]
Bit of cludge - to get `<commentary/>` to appear, we need to add `commentary="yes"` to the `<stringparams/>` for each target in `project.ptx`. We should still use `component="instructor"` as well (which I did not do above) to show/hide commentary as needed, since this stringparam behavior is deprecated and will soon not be necessary.

**Reply by siwelwerd** [reply]
I agree, the red L adds a nice visual flag.

@AbbyANoble @kathypinzon @tdegeorge  What do y'all think?

**Reply by siwelwerd** [reply]
>Bit of cludge - to get <commentary/> to appear, we need to add commentary="yes" to the <stringparams/> for each target in project.ptx.

Just the two targets for instructor versions though right?  Maybe you can drop that in when you review #132.

**Reply by StevenClontz** [reply]
> Just the two targets for instructor versions though right?

Yeah that's fine - once commentary goes live without the stringparam setting, having `component="instructor"` will keep things working the right way.

**Reply by AbbyANoble** [reply]
I like the red L. Visual cue without being cluttered.

**Reply by kathypinzon** [reply]
I agree, I like the red L as well - it would be even better if it could toggle open/closed like the answer tag does.... not sure if that's a possibility or not.

**Reply by StevenClontz** [reply]
I don't believe it is. :-(

**Reply by siwelwerd** [reply]
I thought we could knowl-ify anything--or am I misunderstanding what is happening [here](https://github.com/TeamBasedInquiryLearning/precalculus/blob/b263c687a1e6ed8a6f9abf017c9f33fbcd7eaaad/publication/publication.ptx#L75)?

**Reply by StevenClontz** [reply]
Oh, for whatever reason I didn't think of that working because these things aren't numbered, but maybe it could?

**Reply by StevenClontz** [reply]
That list I believe is exhaustive and I believe doesn't include the commentary element.

**Reply by siwelwerd** [reply]
Darn.  I think it's okay to have instructor notes always displayed in the instructor version though. If someone doesn't want to look at them, they can read the student version.

**Comment by StevenClontz** [comment]
> Maybe you can drop that in when you review #132.

Done.

**Comment by siwelwerd** [comment]
Added example code to our [style guide](https://github.com/TeamBasedInquiryLearning/wiki/wiki/Style-Guide) for how to implement this, but duplicating here for posterity (and easier copypasta)

```
<commentary component="instructor">
  <title>Instructor Note</title>
  <p>Your note goes here.</p>
</commentary>
```
